--- /Volumes/files/htdocs/alloy/trunk/lib/yui/build/io/io.js	2009-08-27 16:14:58.000000000 -0300
+++ /Volumes/files/htdocs/alloy/trunk/lib/yui_/build/io/io.js	2009-08-27 21:52:07.000000000 -0300
@@ -210,7 +210,7 @@
 
    		o.c.onreadystatechange = function() { _readyState(o, c); };
    		try {
-			o.c.open(m, uri, true);
+			o.c.open(m, uri, c.async === false ? false : true);
    		}
    		catch(e0){
 			if (c.xdr) {
@@ -239,6 +239,11 @@
 		}
 
    		_ioStart(o.id, c);
+
+		if (c.async === false) {
+			_readyState(o, c);
+		}
+
    		// If config.timeout is defined, and the request is standard XHR,
    		// initialize timeout polling.
    		if (c.timeout) {
@@ -630,16 +635,22 @@
 	* @return void
    	*/
    	function _readyState(o, c) {
-   		if (o.c.readyState === 4) {
+   		if (o.c && o.c.readyState === 4) {
    			if (c.timeout) {
    				_clearTimeout(o.id);
    			}
 
-   			w.setTimeout(
-   				function() {
-   					_ioComplete(o, c);
-   					_handleResponse(o, c);
-   				}, 0);
+			var complete = function() {
+				_ioComplete(o, c);
+				_handleResponse(o, c);
+			};
+
+			if (c.async === false) {
+				complete();
+			}
+			else {
+				w.setTimeout(complete, 0);
+			}
    		}
    	}
 
